spec/lib/site_settings/deprecated_settings_spec.rb

```diff
@@ -0,0 +1,223 @@
+          expect(SiteSetting.use_https).to eq(true)
+          expect(SiteSetting.use_https?).to eq(true)
+        end
+      expect(logger.warnings.count).to eq(2)
+
+      logger = track_log_messages { SiteSetting.use_https(warn: false) }
+      expect(logger.warnings.count).to eq(0)
+    end
+  end
+
+  describe "when overriding a trust level setting with a group setting" do
+    let(:override) { false }
+    let(:deprecated_test) { "#{Rails.root}/spec/fixtures/site_settings/deprecated_test.yml" }
+
+    before { SiteSetting.load_settings(deprecated_test) }
+
+    context "when getting an old TrustLevelSetting" do
+      before do
+        deprecate_override!(
+          ["min_trust_level_to_allow_invite", "invite_allowed_groups", override, "0.0.1"],
+        )
+      end
+
+      it "uses the minimum trust level from the trust level auto groups in the new group setting" do
+        SiteSetting.invite_allowed_groups =
+          "#{Group::AUTO_GROUPS[:trust_level_3]}|#{Group::AUTO_GROUPS[:trust_level_4]}"
+        expect(SiteSetting.min_trust_level_to_allow_invite).to eq(TrustLevel[3])
+      end
+
+      it "returns nil if there are no trust level auto groups in the new group setting" do
+        SiteSetting.invite_allowed_groups = Fabricate(:group).id.to_s
+        expect(SiteSetting.min_trust_level_to_allow_invite).to eq(nil)
+      end
+
+      it "returns nil if there are only staff and admin auto groups in the new group setting" do
+        SiteSetting.invite_allowed_groups =
+          "#{Group::AUTO_GROUPS[:admins]}|#{Group::AUTO_GROUPS[:staff]}"
+        expect(SiteSetting.min_trust_level_to_allow_invite).to eq(nil)
+      end
+    end
+
+    context "when getting an old TrustLevelAndStaffSetting" do
+      before do
+        deprecate_override!(
+          [
+            "min_trust_level_to_allow_invite_tl_and_staff",
+            "invite_allowed_groups",
+            override,
+            "0.0.1",
+          ],
+          ["min_trust_level_to_allow_invite_tl_and_staff"],
+        )
+      end
+
+      it "returns staff if there are staff and admin auto groups in the new group setting" do
+        SiteSetting.invite_allowed_groups =
+          "#{Group::AUTO_GROUPS[:admins]}|#{Group::AUTO_GROUPS[:staff]}"
+        expect(SiteSetting.min_trust_level_to_allow_invite_tl_and_staff).to eq("staff")
+      end
+
+      it "returns admin if there is only the admin auto group in the new group setting" do
+        SiteSetting.invite_allowed_groups = "#{Group::AUTO_GROUPS[:admins]}"
+        expect(SiteSetting.min_trust_level_to_allow_invite_tl_and_staff).to eq("admin")
+      end
+
+      it "returns the min trust level if the admin auto group as well as lower TL auto groups in the new group setting" do
+        SiteSetting.invite_allowed_groups =
+          "#{Group::AUTO_GROUPS[:admins]}|#{Group::AUTO_GROUPS[:trust_level_3]}"
+        expect(SiteSetting.min_trust_level_to_allow_invite_tl_and_staff).to eq(TrustLevel[3])
+      end
```

Comment:
It is probably worth testing the case where `SiteSetting.invite_allowed_groups` has no auto groups.

Reply:
I guess this would have to be default TL4 as well, since that was the highest you could originally set it to.